### PR TITLE
Add payment module E2E tests for Social Protection

### LINKED ISSUE #28: Add e2e integration tests for Payments
Payment Plan (CRUD)
- Social protection calculation rule
- timesheet calculation rule

Payment Cycle
Payment Point
Payroll

### PR DESCRIPTION
**Pull request by Shahzaibahmad97**
# Description

Adds Cypress E2E test coverage for the **Payments** domain — payment plans, payment cycles, payment points, payrolls, and payroll reconciliation. This PR also includes a foundational refactor of the monolithic `commands.js` into domain-scoped modules and hardens shared support commands (login/logout resilience, MUI date input helper, frontend module config, searcher + form helpers, constants module).

# Type of Change

- [x] Feature
- [ ] Bug fix
- [x] Chore (Refactor, Docs, CI/CD)
- [ ] Other, please specify

## Related Issue(s) / Task(s)

- [Save button Fix for Payment Plan](https://github.com/openimis/openimis-fe-contribution_plan_js/pull/108)
- [Fix undefined Program name on listing](https://github.com/openimis/openimis-fe-contribution_plan_js/pull/109)
- [Payment Point Create Tooltip Key Alignment](https://github.com/openimis/openimis-fe-payroll_js/pull/69)
- [Missing Translations for Payment Cycle Searcher Title and Create Tooltip](https://github.com/openimis/openimis-fe-payment_cycle_js/pull/22)

---

## Refactor: Command Module Split

- [x] Split monolithic `cypress/support/commands.js` into domain-scoped modules under `commands/` (auth, admin, ui, form, searcher, location, programs, registry, tasks, grievance, payments)
- [x] Barrel `commands/index.js` registers all modules
- [x] Add `date.js` helper (`getTodayFormatted`)
- [x] Add shared `constants.js` (`TIMEOUTS`, `CALC_RULES`, `PAYROLL_STATUS`, `PAYMENT_CYCLE_STATUS`)
- [x] Per-folder command reference docs at `cypress/support/commands/COMMANDS.md` and `cypress/support/commands/payments/COMMANDS.md` so spec authors can pick existing helpers instead of re-implementing UI steps

## Refactor: Shared Support Command Enhancements

- [x] Harden `cy.login()` — check DOM for already-authenticated state before entering credentials; clear cookies for clean state; increased timeout to 30s; handle `/front/login`, `/front`, and authenticated redirect states as valid entrypoints
- [x] Harden `cy.logout()` — handle already-logged-out state gracefully (assert login screen if already on it)
- [x] Harden `cy.logoutAdminInterface()` — guard against missing logout button (link or button element)
- [x] Add `enterDateInput` / `selectDateFromCalendar` commands that navigate the MUI DatePicker calendar popup
- [x] Add `setFrontendModuleConfig` command for frontend-layer module configuration
- [x] Add `chooseMuiAutocomplete` generic helper for MUI autocomplete fields
- [x] Add `chooseFirstMuiSelect`, `toggleMuiCheckbox`, `assertMuiInputDisabled`, `assertMuiSelectValue`, `assertMuiAutoComplete`, `assertMuiInputNotEmpty` helpers
- [x] Add generic `saveClick`, `saveAndAwaitJournal`, `assertSave`, `assertSaveDisabled`, `assertSaveEnabled`, `waitForJournalProgress`, `assertJournalFirstEntryContains`, `assertJournalNoFail`, `createClick` form helpers
- [x] Add searcher helpers: `aliasGraphqlQuery` + `awaitSearcherRefresh` (GraphQL-intercept aliasing replaces stale "N X Found" post-search text waits), `resetSearcherFilters`, `assertTableRowVisible`, `assertTableRowNotVisible`, `openRowAction`, `openRowActionIfPresent`
- [x] Add `chooseLocation` + `chooseLocationLevel` location-cascade helpers

## Program / Project Enhancements

- [x] `createProgram` accepts an optional 5th `schema` parameter to set the beneficiary JSON schema during program creation
- [x] Schema validation waits for backend async validation (SVG icon + adornment class check) before saving
- [x] Beneficiary schema (`educated_level`, `able_bodied`, `number_of_children`) passed to individual programs, enabling advanced criteria in payment plan and payroll forms
- [x] **New** `enterProjectTimeEntriesPerBeneficiary(projectPath, daysByText)` — sets per-beneficiary, per-day work-percent values against the timesheet project grid (token-based row matching tolerates first/last name in separate `<td>` cells; re-queries DOM on every cell to avoid stale-handle flakes from MUI table re-renders)

---

### Test Cases (15 tests)

- [x] Validates required fields are enforced before allowing creation (save button disabled)
- [x] Creates a benefit-plan payment plan successfully and verifies it in the list and detail form
- [x] Blocks save when the code is changed to a duplicate value
- [x] Applies advanced criteria from the program JSON schema (field, filter, value, amount)
- [x] Searches payment plans by code and by name
- [x] Edits all editable fields on an existing payment plan and verifies updated values persist
- [x] Adds a new version of a payment plan
- [x] Deletes a payment plan and shows it only in deleted history
- [x] Creates a benefit-plan payment plan for a group-profile program (smoke test)
- [x] Filters payment plans by benefit plan / program
- [x] Applies multiple advanced criteria rows (Educated level + Able bodied)
- [x] Filters payment plans by date range (`dateValidFrom` / `dateValidTo`)
- [x] Resets payment plan filters and restores full list
- [x] Creates a payment plan with timesheet calcrule and Base Day Rate
- [x] Creates a timesheet payment plan without optional Base Day Rate

### Behaviour change — picker now uses program *code*, not name

The Program autocomplete on the payment plan form was failing intermittently because long benefit-plan names get visually truncated/mangled while typing into the MUI input. The picker now searches by the unique 8-char program code, with the human-readable name kept only for display assertions. This means:

- `planData(label, benefitPlanCode, benefitPlanName)` — code is now the primary key
- `fillPaymentPlanForm({ ..., benefitPlanCode, benefitPlanName })` — `benefitPlanCode` is preferred when present, falls back to `benefitPlanName` for legacy callers
- `assertPaymentPlanDetailFields({ ..., benefitPlanCode, benefitPlanName })` — same fallback pattern

### Command Helpers

| Command | Purpose |
|---------|---------|
| `cy.openCreatePaymentPlan()` | Navigate to payment plan list and click create |
| `cy.fillPaymentPlanForm(...)` | Fill type, code, name, calculation rule, calculation params, benefit plan (by code or name), dates, advanced criteria |
| `cy.savePaymentPlan(mutationLabel?)` | Click save and await async journal |
| `cy.createPaymentPlan(...)` | End-to-end: open → fill → save → assert no failure journal entry |
| `cy.filterPaymentPlans(...)` | Filter by code, name, dateValidFrom, dateValidTo, showDeleted, showHistory |
| `cy.resetPaymentPlanFilters()` | Click Reset and wait for the searcher to refresh |
| `cy.assertPaymentPlanDetailFields(...)` | Assert form values after reopening (code-or-name match) |
| `cy.assertPaymentPlanRowVisible(...)` / `cy.assertPaymentPlanRowNotVisible(...)` | Row presence assertions |
| `cy.openPaymentPlanForEditFromList(...)` | Filter → click Edit |
| `cy.replacePaymentPlanFromList(...)` | Filter → click Add New Version |
| `cy.deletePaymentPlan(...)` | Filter → click Delete → confirm dialog → verify journal |

---

### Test Cases (13 tests)

- [x] Validates required fields are enforced before allowing creation
- [x] Creates a PENDING payment cycle successfully
- [x] Searches payment cycles by code
- [x] Views payment cycle details from the list (eye icon)
- [x] Creates an ACTIVE payment cycle via task approval and verifies all fields
- [x] Creates a SUSPENDED payment cycle
- [x] Blocks save when the code is changed to a duplicate value
- [x] Filters payment cycles by status
- [x] Verifies read-only / detail-page behavior
- [x] Creates a payment cycle and immediately searches for it (list refresh)
- [x] Resets payment cycle filters and restores full list
- [x] Filters payment cycles by date range (Date From / Date To)
- [x] Payroll-form `PaymentCyclePicker` excludes non-ACTIVE cycles (asserts a PENDING cycle never appears in the autocomplete options on the payroll create form, since the picker queries `paymentCycle(status: ACTIVE)`)

### Command Helpers

| Command | Purpose |
|---------|---------|
| `cy.openCreatePaymentCycle()` | Navigate to payment cycle list and click create |
| `cy.fillPaymentCycleForm(...)` | Fill code, start date, end date, status |
| `cy.savePaymentCycle()` | PENDING/SUSPENDED redirects to detail URL; ACTIVE dismisses the `coreAlert` dialog and returns to list |
| `cy.createPaymentCycle(...)` | End-to-end: open → fill → save |
| `cy.filterPaymentCycles({ code, status, dateFrom, dateTo })` | Filter by code, status, **and date range** (date params added in this PR) |
| `cy.resetPaymentCycleFilters()` | Click Reset and wait for the searcher to refresh |
| `cy.assertPaymentCycleDetailFields(...)` | Assert form values on the detail page |
| `cy.assertPaymentCycleRowVisible(...)` / `cy.assertPaymentCycleRowNotVisible(...)` | Row presence assertions |
| `cy.openPaymentCycleForViewFromList(...)` | Filter → click View (eye icon) |

---

### Test Cases (16 tests, all passing)

Validates required fields, creates with full location hierarchy, verifies detail page, filters by name / PPM / Region / Region+District / Region+District+Municipality / full hierarchy, wrong-Region-no-match, cascading Region→District options, reset filters, view via eye icon, edits name, deletes from list, deletes from detail page.

### Command Helpers

| Command | Purpose |
|---------|---------|
| `cy.openCreatePaymentPoint()`, `cy.fillPaymentPointForm(...)`, `cy.savePaymentPoint()`, `cy.createPaymentPoint(...)` | Create flow |
| `cy.filterPaymentPoints(...)`, `cy.resetPaymentPointFilters()` | List filter / reset |
| `cy.assertPaymentPointDetailFields(...)`, `cy.assertPaymentPointRowVisible(...)`, `cy.assertPaymentPointRowNotVisible(...)` | Assertions |
| `cy.openPaymentPointForViewFromList(...)`, `cy.deletePaymentPointFromList(...)` | Row actions |

---

### Test Cases (12 tests)

Required-field validation, create + assert, search, view, delete (with `payroll_delete` task approval), pending list visibility, PENDING APPROVAL status assertion, summary dialog open/close, payment plan → cycle → payroll integration smoke, status filter, reset filters, timesheet-calcrule end-to-end.

### Command Helpers (existing)

| Command | Purpose |
|---------|---------|
| `cy.openCreatePayroll()`, `cy.fillPayrollForm(...)`, `cy.savePayroll()`, `cy.createPayroll(...)` | Create flow |
| `cy.filterPayrolls(...)`, `cy.resetPayrollFilters()` | List / pending list filter |
| `cy.assertPayrollDetailFields(...)`, `cy.assertPayrollRowVisible(...)`, `cy.assertPayrollRowNotVisible(...)` | Assertions |
| `cy.openPayrollForViewFromList(...)`, `cy.deletePayrollFromList(...)` | Row actions |
| `cy.openPayrollPendingSummary(...)` | Open reconciliation summary dialog from `/front/payrollsPending` |
| `cy.approveTaskFromList(...)`, `cy.approveLatestPaymentCycleTask()`, `cy.ensurePermissiveTaskGroup()`, `cy.ensurePaymentCycleTaskGroup(...)` | Maker-checker support |

---

## Payroll Reconciliation — `payroll-reconciliation.cy.js`

End-to-end coverage of the reconciliation state machine for both calc-rule families. Each test takes a payroll from `PENDING_APPROVAL` through `APPROVE_FOR_PAYMENT` to `RECONCILED`, including UI-driven CSV download, edit, and re-upload, plus the **Approve and Close** dialog action that creates the `payroll_reconciliation` task whose approval gates the final RECONCILED transition.

### Test Cases (4 implemented)

**Cash transfer / social-protection calcrule:**

- [x] **Full reconciliation** — creates a benefit-plan program (Fixed amount = 250), enrolls 4 individual beneficiaries, sets up an ACTIVE payment cycle and a `social_protection`-calcrule payment plan with `StrategyOfflinePayment`, creates a payroll, approves the `accept_payroll` task, downloads the reconciliation CSV from the UI, marks every row Paid=Yes with a unique receipt, re-uploads through the dialog, asserts the per-beneficiary amounts on the summary dialog, clicks **Approve and Close**, approves the resulting `payroll_reconciliation` task, and asserts payroll status = RECONCILED.

**Timesheet / base-day-rate calcrule:**

- [x] **Per-beneficiary day patterns drive amount calculation** — creates a `timesheet`-calcrule program (Base Day Rate = 100) with 4 beneficiaries assigned to a project; uses `enterProjectTimeEntriesPerBeneficiary` to seed different day patterns per individual: full days (10), partial days (4 + half-days), zero days, mixed half-days. Verifies each beneficiary's `Amount = effective_days × BDR` on the summary dialog (e.g., 5 days = 500, 10 days = 1000, mixed half-days = 400, etc.).
- [x] **Stage-1 partial reconciliation** — uploads a CSV marking only some rows Paid=Yes, asserts the per-row BenefitConsumption status changes are reflected on re-download.
- [x] **Stage-2 full reconciliation + close** — re-downloads, marks all remaining rows Paid=Yes, re-uploads, then runs the **Approve and Close** + `payroll_reconciliation` task approval flow, and asserts payroll status = RECONCILED. (This stage previously appeared to be "missing" because the close+approve sequence was buried in nested `.then` blocks; it is now lifted to outer scope so the flow mirrors the cash-transfer test.)

### Command Helpers

| Command | Purpose |
|---------|---------|
| `cy.parseReconciliationCsv(text)` | Pure parser: returns `{ headers, rows }` from a reconciliation CSV string (handles the BE's 11-column layout) |
| `cy.buildReconciliationCsv(rows, headers?)` | Pure builder: serialises `{ ... }` rows back into CSV text using `RECONCILIATION_CSV_HEADERS` order |
| `cy.downloadReconciliationFromUI()` | Clicks the Download button on the payroll detail page, intercepts `GET /api/payroll/csv_reconciliation/`, returns `{ headers, rows, csvText }` |
| `cy.uploadReconciliationFromUI(payrollName, csvText)` | Writes the CSV to `cypress/fixtures/_generated/recon_<safe>_<timestamp>.csv`, opens the **Upload Payment Data** dialog, attaches the file, intercepts `POST /api/payroll/import_payroll_reconciliation/`, surfaces backend error bodies on non-2xx responses. The timestamp suffix avoids the BE's "File already exists at the specified path" rejection on re-upload |
| `cy.approveAcceptPayrollTask(payrollName)` | Approves the `accept_payroll` task that transitions PENDING_APPROVAL → APPROVE_FOR_PAYMENT |
| `cy.approveAndClosePayrollFromSummary(payrollName)` | From `/front/payrollsApproved`, opens the row's summary dialog and clicks **Approve and Close** (this is the action that creates the `payroll_reconciliation` task — the bit the original test design missed) |
| `cy.approvePayrollReconciliationTask(payrollName)` | Approves the resulting `payroll_reconciliation` task that finally transitions APPROVE_FOR_PAYMENT → RECONCILED |
| `cy.assertReconciliationSummary(payrollName, { selected, total, totalAmount, totalDelivered })` | Scopes inside `[role="dialog"]` and asserts the four summary metrics (uses `.should('exist')` not `be.visible` because the dialog cards exist in DOM but aren't reported as visible by headless Electron) |

### Reconciliation flow notes (for future contributors)

- CSV upload alone only flips per-row `BenefitConsumption.status` — it does **not** transition the payroll itself. The payroll → RECONCILED transition is gated on a user clicking **Approve and Close** on the summary dialog, which creates the `payroll_reconciliation` task. Approving that task is what flips the payroll's status.
- `RELAY_CONNECTION_MAX_LIMIT=100` — beneficiary GraphQL queries must use `first: 100`.
- The CSV "Payroll Status" column carries the raw enum (`APPROVE_FOR_PAYMENT`, with underscores), not the UI label (`APPROVE FOR PAYMENT`). Compare against the literal enum.

---

## Demo

_To be added after test runs are recorded._

## Checklist

- [x] New Cypress commands follow the domain-module pattern (`commands/<domain>.commands.js`)
- [x] Tests clean up created data in `after()` hooks (payment plans, programs, payment points, payrolls where applicable)
- [x] Login/logout commands handle edge cases (already logged in/out)
- [x] Program creation supports beneficiary schema for advanced criteria
- [x] Payroll spec and command helpers added
- [x] Payment point spec and command helpers added
- [x] Payment plan fully green in targeted Cypress run
- [x] Payment cycle fully green in targeted Cypress run
- [x] Payment point fully green in targeted Cypress run (16/16)
- [x] Payroll fully green in targeted Cypress run
- [x] Payroll reconciliation fully green in targeted Cypress run (4/4)
- [x] Per-folder command reference docs (`COMMANDS.md`) added under `cypress/support/commands/` and `cypress/support/commands/payments/` so future spec authors can pick existing helpers instead of re-implementing UI flows
- [ ] Cypress recordings/screenshots attached

### CODE REVIEW

File: cypress/e2e/payment-plan.cy.js [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+
+  const planData = (label, benefitPlanCode, benefitPlanName) => {
+    const timestamp = getTimestamp();
+    const uniqueCodePart = `${Date.now().toString().slice(-5)}${Math.random().toString(36).slice(2, 4).toUpperCase()}`;
+
+    return {
+      code: `E2EPP${uniqueCodePart}`,
+      name: `E2E Payment Plan ${label} ${timestamp}`,
+      benefitPlanCode,
+      benefitPlanName,
```
[the remark this answers is not in the document]
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-plan.cy.js:56]: The 8-char limit applies to benefit-plan / program codes, not payment-plan codes. `PaymentPlan.code` is `max_length=255`. The generated codes here are 12 chars and well within limits — no change needed.

File: cypress/support/commands/tasks.commands.js [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    // and is used as the Entity filter value on /front/allTasks to avoid scanning
+    // pages of accumulated tasks.
+    const filterText = [...texts].sort((a, b) => b.length - a.length)[0];
+
+    const waitForTaskRow = (attempt = 1) => {
+      cy.visit('/front/allTasks');
+      cy.contains('Tasks Found', { timeout: TIMEOUTS.BACKEND_VALIDATION });
+      cy.enterMuiInput('Entity', filterText);
+      cy.aliasGraphqlQuery('task(', `taskSearch_${attempt}`);
+      cy.contains('button', 'Search').click();
```
[the remark this answers is not in the document]
Comment (Shahzaibahmad97) [reply cypress/support/commands/tasks.commands.js]: updated this to use the capitalized url path as this was the one used by openimis. /front/AllTasks

File: cypress/support/commands/tasks.commands.js [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    clickApproveAndConfirm();
+  });
+
+  // Approve the most recently created PaymentCycleService task.
+  Cypress.Commands.add('approveLatestPaymentCycleTask', () => {
+    cy.visit('/front/allTasks');
+    cy.contains('Tasks Found', { timeout: TIMEOUTS.BACKEND_VALIDATION });
+
+    cy.contains('table tbody tr', 'PaymentCycleService', { timeout: TIMEOUTS.BACKEND_VALIDATION })
+      .first()
```
[the remark this answers is not in the document]
Comment (Shahzaibahmad97) [reply cypress/support/commands/tasks.commands.js:182]: updated this to use the capitalized url path as this was the one used by openimis. /front/AllTasks
even though the test cases weren't failing because react would route by default to the correct URL and it wasn't case sensitive.

File: cypress/e2e/payment-cycle.cy.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+    const date = new Date();
+    date.setDate(date.getDate() + days);
+    const day = String(date.getDate()).padStart(2, '0');
+    const month = String(date.getMonth() + 1).padStart(2, '0');
+    const year = date.getFullYear();
+    return `${day}-${month}-${year}`;
+  };
+
+  const cycleData = () => {
+    const uniquePart = `${Date.now().toString().slice(-5)}${Math.random().toString(36).slice(2, 4).toUpperCase()}`;
```
Comment (weilu) [inline cypress/e2e/payment-cycle.cy.js]: Consider using `getTimestamp`? https://github.com/openimis/openimis-dist_dkr/blob/develop/cypress/support/utils.js

File: cypress/e2e/payment-cycle.cy.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+    cy.assertSaveDisabled();
+  });
+
+  it('creates a PENDING payment cycle successfully', () => {
+    const cycle = cycleData();
+
+    cy.createPaymentCycle(cycle);
+
+    cy.filterPaymentCycles({ code: cycle.code });
+    cy.assertPaymentCycleRowVisible({ code: cycle.code });
```
Comment (weilu) [inline cypress/e2e/payment-cycle.cy.js]: assert status as well? `cy.assertPaymentCycleRowVisible({ code: cycle.code, status: 'PENDING' });`

File: cypress/support/commands/form.commands.js [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+    cy.get('ul.MuiList-root li').first().should('contain', text);
+  });
+
+  // Assert that the first journal entry does NOT contain the given prefix.
+  // Typical use: assertJournalNoFail('Failed to create').
+  Cypress.Commands.add('assertJournalNoFail', (prefix = 'Failed to') => {
+    cy.get('ul.MuiList-root li').first().should('not.contain', prefix);
+  });
+
+  Cypress.Commands.add('createClick', (createTitle = 'Create') => {
```
Comment (weilu) [inline cypress/support/commands/form.commands.js]: rename to `clickCreate` instead for readability?

File: cypress/e2e/payment-cycle.cy.js [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+
+  it('validates required fields before allowing payment cycle creation', () => {
+    cy.openCreatePaymentCycle();
+    cy.assertSaveDisabled();
+  });
+
+  it('creates a PENDING payment cycle successfully', () => {
+    const cycle = cycleData();
+
+    cy.createPaymentCycle(cycle);
```
Comment (weilu) [inline cypress/e2e/payment-cycle.cy.js]: Since the test description states "PENDING" consider passing `status = 'PENDING'` explicitly for the createPaymentCycle call

File: cypress/e2e/payment-cycle.cy.js [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+    cy.get('body').then(($body) => {
+      const options = $body.find('[role="listbox"] li, [role="presentation"] li').toArray();
+      const pendingVisible = options.some((li) => li.innerText.includes(pendingCycle.code));
+      expect(
+        pendingVisible,
+        `PENDING cycle ${pendingCycle.code} must not appear in the ACTIVE-filtered picker`,
+      ).to.equal(false);
+    });
+  });
+});
```
Comment (weilu) [inline cypress/e2e/payment-cycle.cy.js:228]: @Shahzaibahmad97 consider organizing the tests by action type: e.g. grouping all search & filter tests, followed by creation tests together, followed by updates.

File: cypress/e2e/payment-plan.cy.js [hunk 9]
```diff
@@ -0,0 +1,409 @@
...
+    cy.contains('Payment Plans Found');
+
+    cy.enterMuiInput('Code', 'FAKE_CODE');
+    cy.enterMuiInput('Name', 'FAKE_NAME');
+
+    cy.resetPaymentPlanFilters();
+
+    cy.assertMuiInput('Code', '');
+    cy.assertMuiInput('Name', '');
+  });
```
Comment (weilu) [inline cypress/e2e/payment-plan.cy.js]: Consider making this more substantial (actually checking the list is restored) & folding this test to one of the other more substantial search/filter tests e.g. reset the filters at the end then check that the total count restores to the default count before any filter is applied.

File: cypress/e2e/payment-plan.cy.js [hunk 10]
```diff
@@ -0,0 +1,409 @@
...
+    cy.assertPaymentPlanDetailFields({
+      code: paymentPlan.code,
+      name: paymentPlan.name,
+      dateValidFrom: paymentPlan.dateValidFrom,
+      calculationRule: CALC_RULES.TIMESHEET,
+      benefitPlanCode: timesheetProgramCode,
+      benefitPlanName: timesheetProgramName,
+    });
+  });
+});
```
Comment (weilu) [inline cypress/e2e/payment-plan.cy.js:436]: Similarly to my comment in the previous file – consider organizing the tests by action type: e.g. grouping all search & filter tests, followed by creation tests together, followed by updates.

File: cypress/e2e/payment-cycle.cy.js (hunk 4 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-cycle.cy.js]: Done in c740aad — `cycleData()` / `planData()` now build the code from the shared `getTimestamp()` util plus a per-run counter. The counter replaces `Math.random()`, which also cleared the SonarCloud PRNG hotspot (S2245) on these lines.

File: cypress/e2e/payment-cycle.cy.js (hunk 5 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-cycle.cy.js]: Done in 3c9fb05. The Status column renders a read-only `PaymentCycleStatusPicker` as an `<input value="PENDING">`, so `cy.contains` (which matches element text, not input values) couldn't see it. `assertPaymentCycleRowVisible` now asserts the status against the row's input values, and the PENDING / ACTIVE / SUSPENDED creation tests pass the expected status.

File: cypress/support/commands/form.commands.js (hunk 6 above)
Comment (Shahzaibahmad97) [reply cypress/support/commands/form.commands.js]: Done in c740aad — renamed the whole family to `clickCreate` / `clickSave` (was `createClick` / `saveClick`), updated all call sites and the COMMANDS.md reference.

File: cypress/e2e/payment-cycle.cy.js (hunk 7 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-cycle.cy.js]: `cycleData()` already sets `status: 'PENDING'` on the object passed to `createPaymentCycle`, so PENDING is passed explicitly. With the row assertion now also verifying status (your other comment, 3c9fb05), the PENDING state is both passed and asserted.

File: cypress/e2e/payment-cycle.cy.js (hunk 8 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-cycle.cy.js:228]: Done in c740aad — regrouped the payment-cycle tests by action type: Search & filter → Creation → Detail & update (with section header comments).

File: cypress/e2e/payment-plan.cy.js (hunk 9 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-plan.cy.js]: Done in c740aad — the test is now substantial: it captures the default "N Payment Plans Found" count, applies a narrowing filter and asserts the count drops, then resets and asserts both that the inputs clear and the count returns to the default.

File: cypress/e2e/payment-plan.cy.js (hunk 10 above)
Comment (Shahzaibahmad97) [reply cypress/e2e/payment-plan.cy.js:436]: Done in c740aad — regrouped the payment-plan tests by action type: Search & filter → Creation → Updates (with section header comments).

**Review by weilu — APPROVED** [review]
